session.go

```diff
@@ -169,7 +161,7 @@ func (s *Session) Open() (net.Conn, error) {
 // OpenStream is used to create a new stream
 func (s *Session) OpenStream() (*Stream, error) {
 	if s.IsClosed() {
-		return nil, ErrSessionShutdown
+		return nil, s.shutdownErr
```

Comment:
This exposes the connection timeout error. I'm happy to revert this but a bunch of tests were complaining that `Write` was returning `ErrSessionShutdown` instead of `ErrConnectionWriteTimeout` so I just fixed this everywhere.